util/logging/logger.go

```diff
@@ -16,18 +16,31 @@ package logging
 
 import (
 	"fmt"
+	"io"
 	"os"
 	"path/filepath"
+	"sync"
 
 	"github.com/sirupsen/logrus"
 )
 
+// How many bytes before rotating the log file.
+// Note that rotation preserves the old file. The total size occupied
+// by the log files on disk is thus capped at double this value.
+//
+// It is a pkg var instead of a const for easier testing.
+var maxLogFileSizeBytes int64 = 1 << 24 // ~16Mb
+// A commonly used suffix for rotated log files.
+const rotatedSuffix = ".1"
```

Comment:
ok, suppose the log file reached its limit, double or not, and there's a new `log.Printf(...)` coming up. how do you keep it under a certain size?

Reply:
I mean, how can you keep a file under a certain size while writing to it at the same time, cutting off the first line at each write? :)

Reply:
Not at every line. Same as now, just not making the `.1` file: if it reaches double the threshold, you truncate half of the file from the front (instead of moving it to a new file), and keep going.

it would be the same in the end, just in one file instead of two files, if that makes sense?

not a big deal anyway, just wondering if the added complexity in code and possibly confusion about which files to request and send in a support case can be reduced.

Reply:
but i see that this is probably not normal practice, so np to keep as is :)

Reply:
yeah you see, "truncate half of the file from the front" is easier said than done. you'd have to open a new temp file, re-open the original file without "append" flag, seek to size/2, start `io.Copy` from the seek'ed original to this new temp file, piping 16Mb of log, then `mv temp original_file` and finally re-open the log file with "append" flag.

a lot of things can fail along the way, plus all the `log.Printf`s from goroutines will be piling up, blocking the backend, until the above is done... basically, the quickest and safest is to just `mv log another_file; re-open log with truncate to 0`, leveraging filesystem/kernel facilities.

Reply:
there are of course many alternatives, like opening a file with `mmap` (memory-mapped file), but you can imagine how much more complicated it'll get... I think not worth it for just a log file.

yeah, we'll need to update the guides and point out there may be 2 files, log.txt and log.txt.1.

Reply:
and I've just realized what I described above is exactly what happens in `truncateHead` func. compare it to `rotate`.

Reply:
i am convinced!